checkstyle/suppressions.xml

```diff
@@ -233,6 +233,12 @@
     <!-- Executor -->
     <suppress checks="ClassFanOutComplexity|MethodCount"
               files="com[\\/]hazelcast[\\/]executor[\\/]impl[\\/]ExecutorServiceProxy"/>
+    <suppress checks="NPathComplexity"
```

Comment:
Kind of always added it there...

Reply:
this should indeed be moved into annotation.

Reply:
This is also more robust when it comes to refactoring (e.g. the annotation will always be moved together with the class, or if we remove an annotated method, the suppression automatically goes away). There are just a few suppressions which don't seem to work with annotations, even on the class level. Otherwise the long-term goal should be to get rid of this XML file (or if possible the suppressions).